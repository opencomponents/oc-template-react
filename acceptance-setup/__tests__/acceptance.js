/**
 * @testEnvironment jsdom
 */

jasmine.DEFAULT_TIMEOUT_INTERVAL = 100000;

const server = require("../server");
const { cli, Registry } = require("oc");
const path = require("path");
const r = require("request-promise-native");
const jsdom = require("jsdom");
const JSDOM = jsdom.JSDOM;
const fs = require("fs-extra");

jest.unmock("minimal-request");

const registryPort = 3000;
const registryUrl = `http://localhost:${registryPort}/`;
const serverPort = 4000;
const serverUrl = `http://localhost:${serverPort}/`;
const ocComponentPath = path.join(
  __dirname,
  "../../acceptance-components/react-app"
);
let registry;
let testServer;

const semverRegex = /\bv?(?:0|[1-9]\d*)\.(?:0|[1-9]\d*)\.(?:0|[1-9]\d*)(?:-[\da-z-]+(?:\.[\da-z-]+)*)?(?:\+[\da-z-]+(?:\.[\da-z-]+)*)?\b/gi;

beforeAll(done => {
  fs.removeSync(path.join(ocComponentPath, "_package"));
  cli.package(
    {
      componentPath: ocComponentPath
    },
    (err, compiledInfo) => {
      if (err) {
        return done(err);
      }

      registry = new Registry({
        local: true,
        discovery: true,
        verbosity: 0,
        path: path.join(__dirname, "../../acceptance-components"),
        port: registryPort,
        baseUrl: registryUrl,
        env: { name: "local" },
        hotReloading: false,
        templates: [require("../../packages/oc-template-react")]
      });

      registry.start(err => {
        if (err) {
          return done(err);
        }
        testServer = server(serverPort, err => {
          if (err) {
            return done(err);
          }
          done();
        });
      });
    }
  );
});

afterAll(done => {
  testServer.close(() => {
    registry.close(() => {
      fs.removeSync(path.join(ocComponentPath, "_package"));
      done();
    });
  });
});

test("Registry should correctly serve rendered and unrendered components", done => {
  const rendered = r(registryUrl + `react-app/?name=SuperMario`)
    .then(function(body) {
      const bodyVersionless = body.replace(semverRegex, "6.6.6");
      expect(bodyVersionless).toMatchSnapshot();
    })
    .catch(err => expect(err).toBeNull());

  const unrendered = r({
    uri: registryUrl + `react-app/?name=SuperMario`,
    headers: {
      Accept: "application/vnd.oc.unrendered+json"
    }
  })
    .then(function(body) {
      const bodyVersionless = body.replace(semverRegex, "6.6.6");
      expect(bodyVersionless).toMatchSnapshot();
    })
    .catch(err => expect(err).toBeNull());

  Promise.all([rendered, unrendered])
    .then(() => done())
    .catch(err => done(err));
});

test("server-side-side rendering", done => {
  JSDOM.fromURL(serverUrl + `?name=SuperMario`, {})
    .then(dom => {
      const domVersionless = dom.serialize().replace(semverRegex, "6.6.6");
      expect(domVersionless).toMatchSnapshot();
      done();
    })
    .catch(err => done(err));
});

test("client-side-side rendering", done => {
  const virtualConsole = new jsdom.VirtualConsole();
  const mockLog = jest.fn(() => {});
  const cssLog = jest.fn();
  virtualConsole.on("log", mockLog);
  JSDOM.fromURL(registryUrl + `react-app/~preview?name=SuperMario`, {
    resources: "usable",
    runScripts: "dangerously",
    virtualConsole
  })
    .then(dom => {
      dom.window.oc = dom.window.oc || {};
      dom.window.oc.cmd = dom.window.oc.cmd || [];
      dom.window.oc.cmd.push(function(oc) {
        oc.events.on("oc:cssDidMount", (ev, css) => {
          cssLog(css);
        });
      });

      setTimeout(() => {
        expect(mockLog.mock.calls).toMatchSnapshot();
        const nameNode = dom.window.document.getElementById("1");
        expect(nameNode).toMatchSnapshot();
        expect(cssLog.mock.calls).toMatchSnapshot();
        done();
      }, 5000);
    })
    .catch(err => done(err));
});
