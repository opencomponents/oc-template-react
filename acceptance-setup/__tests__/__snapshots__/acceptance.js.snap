// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Registry should correctly serve rendered and unrendered components 1`] = `"{\\"type\\":\\"oc-component-local\\",\\"version\\":\\"6.6.6\\",\\"requestVersion\\":\\"\\",\\"name\\":\\"react-app\\",\\"renderMode\\":\\"unrendered\\",\\"href\\":\\"http://localhost:3000/react-app?name=SuperMario\\",\\"data\\":{\\"reactComponent\\":{,\\"src\\":\\"http://localhost:3000/react-app/6.6.6/static/react-component.js\\",\\"props\\":{\\"name\\":\\"SuperMario\\",\\"_staticPath\\":\\"http://localhost:3000/react-app/6.6.6/static/\\",\\"_baseUrl\\":\\"http://localhost:3000/\\",\\"_componentName\\":\\"react-app\\",\\"_componentVersion\\":\\"6.6.6\\"}}},\\"template\\":{\\"src\\":\\"http://localhost:3000/react-app/6.6.6/static/template.js\\",\\"type\\":\\"oc-template-react\\",}}"`;

exports[`Registry should correctly serve rendered and unrendered components 2`] = `"{\\"type\\":\\"oc-component-local\\",\\"version\\":\\"6.6.6\\",\\"requestVersion\\":\\"\\",\\"name\\":\\"react-app\\",\\"renderMode\\":\\"rendered\\",\\"href\\":\\"http://localhost:3000/react-app?name=SuperMario\\",\\"html\\":\\"<oc-component href=\\\\\\"http://localhost:3000/react-app?name=SuperMario\\\\\\"  data-name=\\\\\\"react-app\\\\\\" data-rendered=\\\\\\"true\\\\\\" data-version=\\\\\\"6.6.6\\\\\\"><div id=\\\\\\"oc-reactRoot-react-app\\\\\\" class=\\\\\\"oc-reactRoot-react-app\\\\\\"><div class=\\\\\\"just-special__3WWFUmyX\\\\\\"><h1 class=\\\\\\"super-special__2fC_7zSu textColor__3g8gQnty\\\\\\" id=\\\\\\"1\\\\\\">Hello <!-- -->SuperMario</h1><p>component name: <!-- -->react-app</p><p>component version: <!-- -->6.6.6</p><p>registry baseUrl: <!-- -->http://localhost:3000/</p><p>component staticPath: <!-- -->http://localhost:3000/react-app/6.6.6/static/</p></div></div><style>.textColor__3g8gQnty{color:green}/*! normalize.css 6.6.6 | MIT License | github.com/necolas/normalize.css */html{line-height:1.15;-ms-text-size-adjust:100%;-webkit-text-size-adjust:100%}body{margin:0}article,aside,footer,header,nav,section{display:block}h1{font-size:2em;margin:.67em 0}figcaption,figure,main{display:block}figure{margin:1em 40px}hr{box-sizing:content-box;height:0;overflow:visible}pre{font-family:monospace,monospace;font-size:1em}a{background-color:transparent;-webkit-text-decoration-skip:objects}abbr[title]{border-bottom:none;text-decoration:underline;-webkit-text-decoration:underline dotted;text-decoration:underline dotted}b,strong{font-weight:inherit}b,strong{font-weight:bolder}code,kbd,samp{font-family:monospace,monospace;font-size:1em}dfn{font-style:italic}mark{background-color:#ff0;color:#000}small{font-size:80%}sub,sup{font-size:75%;line-height:0;position:relative;vertical-align:baseline}sub{bottom:-.25em}sup{top:-.5em}audio,video{display:inline-block}audio:not([controls]){display:none;height:0}img{border-style:none}svg:not(:root){overflow:hidden}button,input,optgroup,select,textarea{font-family:sans-serif;font-size:100%;line-height:1.15;margin:0}button,input{overflow:visible}button,select{text-transform:none}[type=reset],[type=submit],button,html [type=button]{-webkit-appearance:button}[type=button]::-moz-focus-inner,[type=reset]::-moz-focus-inner,[type=submit]::-moz-focus-inner,button::-moz-focus-inner{border-style:none;padding:0}[type=button]:-moz-focusring,[type=reset]:-moz-focusring,[type=submit]:-moz-focusring,button:-moz-focusring{outline:1px dotted ButtonText}fieldset{padding:.35em .75em .625em}legend{box-sizing:border-box;color:inherit;display:table;max-width:100%;padding:0;white-space:normal}progress{display:inline-block;vertical-align:baseline}textarea{overflow:auto}[type=checkbox],[type=radio]{box-sizing:border-box;padding:0}[type=number]::-webkit-inner-spin-button,[type=number]::-webkit-outer-spin-button{height:auto}[type=search]{-webkit-appearance:textfield;outline-offset:-2px}[type=search]::-webkit-search-cancel-button,[type=search]::-webkit-search-decoration{-webkit-appearance:none}::-webkit-file-upload-button{-webkit-appearance:button;font:inherit}details,menu{display:block}summary{display:list-item}canvas{display:inline-block}template{display:none}[hidden]{display:none}.just-special__3WWFUmyX{background:#db7093;color:#fff;display:flex}/*! Something \\\\\\"doublyQueoted\\\\\\" and \\\\\\"singlyWuoted\\\\\\" */.super-special__2fC_7zSu{background-color:pink}</style><script>window.oc = window.oc || {};oc.cmd = oc.cmd || [];oc.cmd.push(function(oc){oc.events.fire('oc:cssDidMount', '.textColor__3g8gQnty{color:green}/*! normalize.css 6.6.6 | MIT License | github.com/necolas/normalize.css */html{line-height:1.15;-ms-text-size-adjust:100%;-webkit-text-size-adjust:100%}body{margin:0}article,aside,footer,header,nav,section{display:block}h1{font-size:2em;margin:.67em 0}figcaption,figure,main{display:block}figure{margin:1em 40px}hr{box-sizing:content-box;height:0;overflow:visible}pre{font-family:monospace,monospace;font-size:1em}a{background-color:transparent;-webkit-text-decoration-skip:objects}abbr[title]{border-bottom:none;text-decoration:underline;-webkit-text-decoration:underline dotted;text-decoration:underline dotted}b,strong{font-weight:inherit}b,strong{font-weight:bolder}code,kbd,samp{font-family:monospace,monospace;font-size:1em}dfn{font-style:italic}mark{background-color:#ff0;color:#000}small{font-size:80%}sub,sup{font-size:75%;line-height:0;position:relative;vertical-align:baseline}sub{bottom:-.25em}sup{top:-.5em}audio,video{display:inline-block}audio:not([controls]){display:none;height:0}img{border-style:none}svg:not(:root){overflow:hidden}button,input,optgroup,select,textarea{font-family:sans-serif;font-size:100%;line-height:1.15;margin:0}button,input{overflow:visible}button,select{text-transform:none}[type=reset],[type=submit],button,html [type=button]{-webkit-appearance:button}[type=button]::-moz-focus-inner,[type=reset]::-moz-focus-inner,[type=submit]::-moz-focus-inner,button::-moz-focus-inner{border-style:none;padding:0}[type=button]:-moz-focusring,[type=reset]:-moz-focusring,[type=submit]:-moz-focusring,button:-moz-focusring{outline:1px dotted ButtonText}fieldset{padding:.35em .75em .625em}legend{box-sizing:border-box;color:inherit;display:table;max-width:100%;padding:0;white-space:normal}progress{display:inline-block;vertical-align:baseline}textarea{overflow:auto}[type=checkbox],[type=radio]{box-sizing:border-box;padding:0}[type=number]::-webkit-inner-spin-button,[type=number]::-webkit-outer-spin-button{height:auto}[type=search]{-webkit-appearance:textfield;outline-offset:-2px}[type=search]::-webkit-search-cancel-button,[type=search]::-webkit-search-decoration{-webkit-appearance:none}::-webkit-file-upload-button{-webkit-appearance:button;font:inherit}details,menu{display:block}summary{display:list-item}canvas{display:inline-block}template{display:none}[hidden]{display:none}.just-special__3WWFUmyX{background:#db7093;color:#fff;display:flex}/*! Something \\\\\\"doublyQueoted\\\\\\" and \\\\\\"singlyWuoted\\\\\\" */.super-special__2fC_7zSu{background-color:pink}');oc.requireSeries([{\\\\\\"global\\\\\\":[\\\\\\"Object\\\\\\",\\\\\\"assign\\\\\\"],\\\\\\"url\\\\\\":\\\\\\"https://unpkg.com/es6-object-assign@6.6.6/dist/object-assign-auto.min.js\\\\\\",\\\\\\"name\\\\\\":\\\\\\"Object.assign\\\\\\"},{\\\\\\"global\\\\\\":\\\\\\"PropTypes\\\\\\",\\\\\\"url\\\\\\":\\\\\\"https://unpkg.com/prop-types@6.6.6/prop-types.min.js\\\\\\",\\\\\\"name\\\\\\":\\\\\\"prop-types\\\\\\"},{\\\\\\"global\\\\\\":\\\\\\"React\\\\\\",\\\\\\"url\\\\\\":\\\\\\"https://unpkg.com/react@6.6.6/umd/react.production.min.js\\\\\\",\\\\\\"name\\\\\\":\\\\\\"react\\\\\\"},{\\\\\\"global\\\\\\":\\\\\\"ReactDOM\\\\\\",\\\\\\"url\\\\\\":\\\\\\"https://unpkg.com/react-dom@6.6.6/umd/react-dom.production.min.js\\\\\\",\\\\\\"name\\\\\\":\\\\\\"react-dom\\\\\\"}], function(){oc.require([\\"oc\\", \\"reactComponents\\", \\"dummyContent\\"],\\\\\\"http://localhost:3000/react-app/6.6.6/static/react-component.js\\\\\\",function(ReactComponent){var targetNode = document.getElementById(\\\\\\"oc-reactRoot-react-app\\\\\\");targetNode.setAttribute(\\\\\\"id\\\\\\",\\\\\\"\\\\\\");ReactDOM.hydrate(React.createElement(ReactComponent,{\\\\\\"name\\\\\\":\\\\\\"SuperMario\\\\\\",\\\\\\"_staticPath\\\\\\":\\\\\\"http://localhost:3000/react-app/6.6.6/static/\\\\\\",\\\\\\"_baseUrl\\\\\\":\\\\\\"http://localhost:3000/\\\\\\",\\\\\\"_componentName\\\\\\":\\\\\\"react-app\\\\\\",\\\\\\"_componentVersion\\\\\\":\\\\\\"6.6.6\\\\\\"}),targetNode);});});});</script><script>window.oc=window.oc||{};oc.renderedComponents=oc.renderedComponents||{};oc.renderedComponents[\\\\\\"react-app\\\\\\"]=\\\\\\"6.6.6\\\\\\";</script></oc-component>\\"}"`;

exports[`client-side-side rendering 1`] = `
Array [
  Array [
    Object {
      "name": "SuperMario",
    },
  ],
]
`;

exports[`client-side-side rendering 2`] = `
<h1
  class="super-special__2fC_7zSu textColor__3g8gQnty"
  id="1"
>
  Hello 
  Pippo
</h1>
`;

exports[`client-side-side rendering 3`] = `
Array [
  Array [
    ".textColor__3g8gQnty{color:green}/*! normalize.css v7.0.0 | MIT License | github.com/necolas/normalize.css */html{line-height:1.15;-ms-text-size-adjust:100%;-webkit-text-size-adjust:100%}body{margin:0}article,aside,footer,header,nav,section{display:block}h1{font-size:2em;margin:.67em 0}figcaption,figure,main{display:block}figure{margin:1em 40px}hr{box-sizing:content-box;height:0;overflow:visible}pre{font-family:monospace,monospace;font-size:1em}a{background-color:transparent;-webkit-text-decoration-skip:objects}abbr[title]{border-bottom:none;text-decoration:underline;-webkit-text-decoration:underline dotted;text-decoration:underline dotted}b,strong{font-weight:inherit}b,strong{font-weight:bolder}code,kbd,samp{font-family:monospace,monospace;font-size:1em}dfn{font-style:italic}mark{background-color:#ff0;color:#000}small{font-size:80%}sub,sup{font-size:75%;line-height:0;position:relative;vertical-align:baseline}sub{bottom:-.25em}sup{top:-.5em}audio,video{display:inline-block}audio:not([controls]){display:none;height:0}img{border-style:none}svg:not(:root){overflow:hidden}button,input,optgroup,select,textarea{font-family:sans-serif;font-size:100%;line-height:1.15;margin:0}button,input{overflow:visible}button,select{text-transform:none}[type=reset],[type=submit],button,html [type=button]{-webkit-appearance:button}[type=button]::-moz-focus-inner,[type=reset]::-moz-focus-inner,[type=submit]::-moz-focus-inner,button::-moz-focus-inner{border-style:none;padding:0}[type=button]:-moz-focusring,[type=reset]:-moz-focusring,[type=submit]:-moz-focusring,button:-moz-focusring{outline:1px dotted ButtonText}fieldset{padding:.35em .75em .625em}legend{box-sizing:border-box;color:inherit;display:table;max-width:100%;padding:0;white-space:normal}progress{display:inline-block;vertical-align:baseline}textarea{overflow:auto}[type=checkbox],[type=radio]{box-sizing:border-box;padding:0}[type=number]::-webkit-inner-spin-button,[type=number]::-webkit-outer-spin-button{height:auto}[type=search]{-webkit-appearance:textfield;outline-offset:-2px}[type=search]::-webkit-search-cancel-button,[type=search]::-webkit-search-decoration{-webkit-appearance:none}::-webkit-file-upload-button{-webkit-appearance:button;font:inherit}details,menu{display:block}summary{display:list-item}canvas{display:inline-block}template{display:none}[hidden]{display:none}.just-special__3WWFUmyX{background:#db7093;color:#fff;display:flex}/*! Something \\"doublyQueoted\\" and \\"singlyWuoted\\" */.super-special__2fC_7zSu{background-color:pink}",
  ],
]
`;

exports[`server-side-side rendering 1`] = `
"<!DOCTYPE html><html><head>
            <title>A page</title>
          </head>
          <body>
          <oc-component href=\\"http://localhost:3000/react-app?name=SuperMario\\" data-name=\\"react-app\\" data-rendered=\\"true\\" data-version=\\"6.6.6\\"><div id=\\"oc-reactRoot-react-app\\" class=\\"oc-reactRoot-react-app\\"><div class=\\"just-special__3WWFUmyX\\"><h1 class=\\"super-special__2fC_7zSu textColor__3g8gQnty\\" id=\\"1\\">Hello <!-- -->SuperMario</h1><p>component name: <!-- -->react-app</p><p>component version: <!-- -->6.6.6</p><p>registry baseUrl: <!-- -->http://localhost:3000/</p><p>component staticPath: <!-- -->http://localhost:3000/react-app/6.6.6/static/</p></div></div><style>.textColor__3g8gQnty{color:green}/*! normalize.css 6.6.6 | MIT License | github.com/necolas/normalize.css */html{line-height:1.15;-ms-text-size-adjust:100%;-webkit-text-size-adjust:100%}body{margin:0}article,aside,footer,header,nav,section{display:block}h1{font-size:2em;margin:.67em 0}figcaption,figure,main{display:block}figure{margin:1em 40px}hr{box-sizing:content-box;height:0;overflow:visible}pre{font-family:monospace,monospace;font-size:1em}a{background-color:transparent;-webkit-text-decoration-skip:objects}abbr[title]{border-bottom:none;text-decoration:underline;-webkit-text-decoration:underline dotted;text-decoration:underline dotted}b,strong{font-weight:inherit}b,strong{font-weight:bolder}code,kbd,samp{font-family:monospace,monospace;font-size:1em}dfn{font-style:italic}mark{background-color:#ff0;color:#000}small{font-size:80%}sub,sup{font-size:75%;line-height:0;position:relative;vertical-align:baseline}sub{bottom:-.25em}sup{top:-.5em}audio,video{display:inline-block}audio:not([controls]){display:none;height:0}img{border-style:none}svg:not(:root){overflow:hidden}button,input,optgroup,select,textarea{font-family:sans-serif;font-size:100%;line-height:1.15;margin:0}button,input{overflow:visible}button,select{text-transform:none}[type=reset],[type=submit],button,html [type=button]{-webkit-appearance:button}[type=button]::-moz-focus-inner,[type=reset]::-moz-focus-inner,[type=submit]::-moz-focus-inner,button::-moz-focus-inner{border-style:none;padding:0}[type=button]:-moz-focusring,[type=reset]:-moz-focusring,[type=submit]:-moz-focusring,button:-moz-focusring{outline:1px dotted ButtonText}fieldset{padding:.35em .75em .625em}legend{box-sizing:border-box;color:inherit;display:table;max-width:100%;padding:0;white-space:normal}progress{display:inline-block;vertical-align:baseline}textarea{overflow:auto}[type=checkbox],[type=radio]{box-sizing:border-box;padding:0}[type=number]::-webkit-inner-spin-button,[type=number]::-webkit-outer-spin-button{height:auto}[type=search]{-webkit-appearance:textfield;outline-offset:-2px}[type=search]::-webkit-search-cancel-button,[type=search]::-webkit-search-decoration{-webkit-appearance:none}::-webkit-file-upload-button{-webkit-appearance:button;font:inherit}details,menu{display:block}summary{display:list-item}canvas{display:inline-block}template{display:none}[hidden]{display:none}.just-special__3WWFUmyX{background:#db7093;color:#fff;display:flex}/*! Something \\"doublyQueoted\\" and \\"singlyWuoted\\" */.super-special__2fC_7zSu{background-color:pink}</style><script>window.oc = window.oc || {};oc.cmd = oc.cmd || [];oc.cmd.push(function(oc){oc.events.fire('oc:cssDidMount', '.textColor__3g8gQnty{color:green}/*! normalize.css 6.6.6 | MIT License | github.com/necolas/normalize.css */html{line-height:1.15;-ms-text-size-adjust:100%;-webkit-text-size-adjust:100%}body{margin:0}article,aside,footer,header,nav,section{display:block}h1{font-size:2em;margin:.67em 0}figcaption,figure,main{display:block}figure{margin:1em 40px}hr{box-sizing:content-box;height:0;overflow:visible}pre{font-family:monospace,monospace;font-size:1em}a{background-color:transparent;-webkit-text-decoration-skip:objects}abbr[title]{border-bottom:none;text-decoration:underline;-webkit-text-decoration:underline dotted;text-decoration:underline dotted}b,strong{font-weight:inherit}b,strong{font-weight:bolder}code,kbd,samp{font-family:monospace,monospace;font-size:1em}dfn{font-style:italic}mark{background-color:#ff0;color:#000}small{font-size:80%}sub,sup{font-size:75%;line-height:0;position:relative;vertical-align:baseline}sub{bottom:-.25em}sup{top:-.5em}audio,video{display:inline-block}audio:not([controls]){display:none;height:0}img{border-style:none}svg:not(:root){overflow:hidden}button,input,optgroup,select,textarea{font-family:sans-serif;font-size:100%;line-height:1.15;margin:0}button,input{overflow:visible}button,select{text-transform:none}[type=reset],[type=submit],button,html [type=button]{-webkit-appearance:button}[type=button]::-moz-focus-inner,[type=reset]::-moz-focus-inner,[type=submit]::-moz-focus-inner,button::-moz-focus-inner{border-style:none;padding:0}[type=button]:-moz-focusring,[type=reset]:-moz-focusring,[type=submit]:-moz-focusring,button:-moz-focusring{outline:1px dotted ButtonText}fieldset{padding:.35em .75em .625em}legend{box-sizing:border-box;color:inherit;display:table;max-width:100%;padding:0;white-space:normal}progress{display:inline-block;vertical-align:baseline}textarea{overflow:auto}[type=checkbox],[type=radio]{box-sizing:border-box;padding:0}[type=number]::-webkit-inner-spin-button,[type=number]::-webkit-outer-spin-button{height:auto}[type=search]{-webkit-appearance:textfield;outline-offset:-2px}[type=search]::-webkit-search-cancel-button,[type=search]::-webkit-search-decoration{-webkit-appearance:none}::-webkit-file-upload-button{-webkit-appearance:button;font:inherit}details,menu{display:block}summary{display:list-item}canvas{display:inline-block}template{display:none}[hidden]{display:none}.just-special__3WWFUmyX{background:#db7093;color:#fff;display:flex}/*! Something \\"doublyQueoted\\" and \\"singlyWuoted\\" */.super-special__2fC_7zSu{background-color:pink}');oc.requireSeries([{\\"global\\":[\\"Object\\",\\"assign\\"],\\"url\\":\\"https://unpkg.com/es6-object-assign@6.6.6/dist/object-assign-auto.min.js\\",\\"name\\":\\"Object.assign\\"},{\\"global\\":\\"PropTypes\\",\\"url\\":\\"https://unpkg.com/prop-types@6.6.6/prop-types.min.js\\",\\"name\\":\\"prop-types\\"},{\\"global\\":\\"React\\",\\"url\\":\\"https://unpkg.com/react@6.6.6/umd/react.production.min.js\\",\\"name\\":\\"react\\"},{\\"global\\":\\"ReactDOM\\",\\"url\\":\\"https://unpkg.com/react-dom@6.6.6/umd/react-dom.production.min.js\\",\\"name\\":\\"react-dom\\"}], function(){oc.require([\\"oc\\", \\"reactComponents\\", \\"dummyContent\\"],\\"http://localhost:3000/react-app/6.6.6/static/react-component.js\\",function(ReactComponent){var targetNode = document.getElementById(\\"oc-reactRoot-react-app\\");targetNode.setAttribute(\\"id\\",\\"\\");ReactDOM.hydrate(React.createElement(ReactComponent,{\\"name\\":\\"SuperMario\\",\\"_staticPath\\":\\"http://localhost:3000/react-app/6.6.6/static/\\",\\"_baseUrl\\":\\"http://localhost:3000/\\",\\"_componentName\\":\\"react-app\\",\\"_componentVersion\\":\\"6.6.6\\"}),targetNode);});});});</script><script>window.oc=window.oc||{};oc.renderedComponents=oc.renderedComponents||{};oc.renderedComponents[\\"react-app\\"]=\\"6.6.6\\";</script></oc-component>
          <script>window.oc=window.oc||{};oc.conf=oc.conf||{};oc.conf.templates=(oc.conf.templates||[]).concat([{\\"type\\":\\"oc-template-es6\\",\\"version\\":\\"6.6.6\\",\\"externals\\":[]},{\\"type\\":\\"oc-template-jade\\",\\"version\\":\\"6.6.6\\",\\"externals\\":[{\\"global\\":\\"jade\\",\\"url\\":\\"https://unpkg.com/jade-legacy@6.6.6/runtime.js\\",\\"name\\":\\"jade\\"}]},{\\"type\\":\\"oc-template-handlebars\\",\\"version\\":\\"6.6.6\\",\\"externals\\":[{\\"global\\":\\"Handlebars\\",\\"url\\":\\"https://unpkg.com/handlebars@6.6.6/dist/handlebars.runtime.min.js\\",\\"name\\":\\"handlebars\\"}]},{\\"type\\":\\"oc-template-react\\",\\"version\\":\\"6.6.6\\",\\"externals\\":[{\\"global\\":[\\"Object\\",\\"assign\\"],\\"url\\":\\"https://unpkg.com/es6-object-assign@6.6.6/dist/object-assign-auto.min.js\\",\\"name\\":\\"Object.assign\\"},{\\"global\\":\\"PropTypes\\",\\"url\\":\\"https://unpkg.com/prop-types@6.6.6/prop-types.min.js\\",\\"name\\":\\"prop-types\\"},{\\"global\\":\\"React\\",\\"url\\":\\"https://unpkg.com/react@6.6.6/umd/react.production.min.js\\",\\"name\\":\\"react\\"},{\\"global\\":\\"ReactDOM\\",\\"url\\":\\"https://unpkg.com/react-dom@6.6.6/umd/react-dom.production.min.js\\",\\"name\\":\\"react-dom\\"}]}]);</script><script src=\\"http://localhost:3000/oc-client/6.6.6/static/src/oc-client.min.js\\" type=\\"text/javascript\\"></script><script>window.oc=window.oc||{};oc.renderedComponents=oc.renderedComponents||{};oc.renderedComponents[\\"oc-client\\"]=\\"6.6.6\\";</script>
          
        
      </body></html>"
`;
